resources/rest-service/cloudify/types/types.yaml

```diff
@@ -29,22 +29,143 @@ node_types:
     derived_from: cloudify.nodes.Root
     properties:
       install_agent:
+        description: |
+          Property for specifying that this Compute node will be
+          running a cloudify agent.
+        default: true
+      remote_execution:
+        description: |
+          Property specifying how will the cloudify agent be installed on
+          the Compute node. If the 'install_agent' property is false, this
+          property is ignored since this node will not be running an agent.
+          If 'install_agent' is true, this property should set to true in
+          order for cloudify to remotely install the agent, and false if the
+          agent installation process takes place outside the context of
+          cloudify (e.g userdata)
         default: true
       cloudify_agent:
         default: {}
       ip:
         default: ''
+      os:
```

Comment:
see where this property is used. adding this property should not be taken lightly as it has backward compatibility issues involved for when we introduce requirements and capabilities.